src/components/entropy/index.js

```diff
@@ -178,13 +179,15 @@ class Entropy extends React.Component {
       </div>
     );
   }
-  requestRecalculationOfBars() {
+  requestRecalculationOfBars(onScreen) {
     /* The <Entropy> component is now responsible for requesting a recalculation of the underlying entropy data
     as this allows us to skip calculations when (e.g.) the panel is not rendered */
     this.props.dispatch((dispatch, getState) => {
       const { entropy, tree } = getState();
       const [bars, maxYVal] = calcEntropyInView(tree.nodes, tree.visibility, entropy.selectedCds, entropy.showCounts);
-      dispatch({type: ENTROPY_DATA, data: bars, maxYVal});
+      const action = {type: ENTROPY_DATA, data: bars, maxYVal};
+      if (onScreen!==undefined) action.onScreen = onScreen;
+      dispatch(action);
```

Comment:
_non-blocking_

> This results in slightly strange behaviour when the entropy panel will
be shown with no bars and they'll be drawn after a slight delay (while
the data's recalculated).

Would it be worth displaying a brief loading text (e.g. "Recalculating entropy") while waiting for the data?

Reply:
Implemented in [3799a98](https://github.com/nextstrain/auspice/pull/1879/commits/3799a9873934684f71cc331e8844c607792dbc3c) - if people don't like it I'm happy to drop it, but I think it's an improvement in and of itself. Note that it's not entirely consistent -- when we change the filters and the tree & entropy panels are in view then we prioritize the tree update and defer entropy recalculations until that's done which leaves the old bars rendered until this time.